src/index.js

```diff
@@ -234,18 +234,21 @@ class Client {
   }
 
   isV2 () {
-    if (!this._version) {
-      this._version = retry(() => fetch(`${this._url}/status/`), 3)()
+    if (!this._cozyVersion) {
+      return retry(() => fetch(`${this._url}/status/`), 3)()
         .then((res) => {
           if (!res.ok) {
             throw new Error('Could not fetch cozy status')
           } else {
             return res.json()
           }
         })
-        .then((status) => status.datasystem !== undefined)
+        .then((status) => {
+          this._cozyVersion = status.datasystem !== undefined ? 2 : 3
+          return this.isV2()
+        })
     }
-    return this._version
+    return Promise.resolve(this._cozyVersion === 2)
```

Comment:
`return` statements seem weird:

- you return a `Promise.resolve` or
- a `retry()` promise

Before those changes the code returns a value. It is not the same type.